xplique/example_based/similar_examples.py

```diff
@@ -0,0 +1,221 @@
+        Labels associated to the examples in the dataset. Indices should match with cases_dataset.
+        `tf.data.Dataset` are assumed to be batched as tensorflow provide no method to verify it.
+        Batch size and cardinality of other datasets should match `cases_dataset`.
+        Be careful, `tf.data.Dataset` are often reshuffled at each iteration, be sure that it is not
+        the case for your dataset, otherwise, examples will not make sense.
+    targets_dataset
+        Targets associated to the cases_dataset for dataset projection,
+        oftentimes the one-hot encoding of a model's predictions. See `projection` for detail.
+        `tf.data.Dataset` are assumed to be batched as tensorflow provide no method to verify it.
+        Batch size and cardinality of other datasets should match `cases_dataset`.
+        Be careful, `tf.data.Dataset` are often reshuffled at each iteration, be sure that it is not
+        the case for your dataset, otherwise, examples will not make sense.
+    k
+        The number of examples to retrieve per input.
+    distance
+        Distance function for examples search. It can be an integer, a string in
+        {"manhattan", "euclidean", "cosine", "chebyshev", "inf"}, or a Callable,
+        by default "euclidean".
+    case_returns
+        String or list of string with the elements to return in `self.explain()`.
+        See the base class returns property for details.
+    batch_size
+        Number of sample treated simultaneously for projection and search.
+        Ignored if `tf.data.Dataset` are provided (those are supposed to be batched).
+    latent_layer
+        Layer used to split the model, the first part will be used for projection and
+        the second to compute the attributions. By default, the model is not split.
+        For such split, the `model` should be a `tf.keras.Model`.
+
+        Layer to target for the outputs (e.g logits or after softmax).
+        If an `int` is provided it will be interpreted as a layer index.
+        If a `string` is provided it will look for the layer name.
+
+        The method as described in the paper apply the separation on the last convolutional layer.
+        To do so, the `"last_conv"` parameter will extract it.
+        Otherwise, `-1` could be used for the last layer before softmax.
+    attribution_method
+        Class of the attribution method to use for projection.
+        It should inherit from `xplique.attributions.base.BlackBoxExplainer`.
+        By default, it computes the gradient to make the Hadamard product in the latent space.
+    attribution_kwargs
+        Parameters to be passed for the construction of the `attribution_method`.
+    """
+    def __init__(
+        self,
+        cases_dataset: Union[tf.data.Dataset, tf.Tensor, np.ndarray],
+        model: tf.keras.Model,
+        targets_dataset: Union[tf.Tensor, np.ndarray],
+        labels_dataset: Optional[Union[tf.Tensor, np.ndarray]] = None,
```

Comment:
To update.

Put in a coherent order.

Targets can be None.

Model can be pytorch (to verify).